### PR TITLE
Replace kotlin extension with view bindings in filesharing views

### PR DESCRIPTION
**Pull request by LunarX**

### CODE REVIEW

File: app/src/main/java/com/infomaniak/drive/ui/fileList/fileShare/FileShareAddUserDialog.kt [hunk 1]
```diff
@@ -45,31 +46,32 @@ import com.infomaniak.lib.core.utils.CoilUtils.simpleImageLoader
...
 class FileShareAddUserDialog : FullScreenBottomSheetDialog() {
 
+    private var binding: FragmentBottomSheetFileShareBinding by safeBinding()
+
     private lateinit var availableUsersAdapter: AvailableShareableItemsAdapter
     private val fileShareViewModel: FileShareViewModel by navGraphViewModels(R.id.fileShareDetailsFragment)
     private val navigationArgs: FileShareAddUserDialogArgs by navArgs()
     private val selectedItems: Share = Share()
 
     private var selectedPermission: Shareable.ShareablePermission = Shareable.ShareablePermission.READ
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/drive/ui/fileList/fileShare/FileShareAddUserDialog.kt:63]: Import `Shareable.`

File: app/src/main/java/com/infomaniak/drive/ui/fileList/fileShare/FileShareAddUserDialog.kt [hunk 2]
```diff
@@ -105,7 +107,7 @@ class FileShareAddUserDialog : FullScreenBottomSheetDialog() {
             selectedPermission = bundle.getParcelable(PERMISSION_BUNDLE_KEY)!!
         }
 
-        shareButton.initProgress(this)
+        shareButton.initProgress(this@FileShareAddUserDialog)
         shareButton.setOnClickListener {
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/drive/ui/fileList/fileShare/FileShareAddUserDialog.kt:111]: Use `shareButton.apply { … }`

File: app/src/main/java/com/infomaniak/drive/ui/fileList/fileShare/FileShareLinkSettingsFragment.kt [hunk 3]
```diff
@@ -32,36 +32,38 @@ import com.infomaniak.drive.MatomoDrive.trackShareRightsEvent
...
 import com.infomaniak.drive.utils.showSnackbar
 import com.infomaniak.drive.views.ShareLinkContainerView.Companion.getTypeName
 import com.infomaniak.lib.core.utils.*
-import kotlinx.android.synthetic.main.fragment_file_share_link_settings.*
 import kotlinx.android.synthetic.main.item_dropbox_settings.expirationDateSwitch
 import java.util.Calendar
 import java.util.Date
 
 class FileShareLinkSettingsFragment : Fragment() {
+    private var binding: FragmentFileShareLinkSettingsBinding by safeBinding()
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/drive/ui/fileList/fileShare/FileShareLinkSettingsFragment.kt:48]:
```suggestion

    private var binding: FragmentFileShareLinkSettingsBinding by safeBinding()
```

**Review by KevinBoulongne — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/infomaniak/drive/ui/fileList/fileShare/FileShareAddUserDialog.kt (hunk 1 above)
Comment (LunarX) [reply app/src/main/java/com/infomaniak/drive/ui/fileList/fileShare/FileShareAddUserDialog.kt:63]: Let's do this kind of improvement in another PR where regroup everything

File: app/src/main/java/com/infomaniak/drive/ui/fileList/fileShare/FileShareAddUserDialog.kt (hunk 2 above)
Comment (LunarX) [reply app/src/main/java/com/infomaniak/drive/ui/fileList/fileShare/FileShareAddUserDialog.kt:111]: Let's do this kind of improvement in another PR where regroup everything

**Review by KevinBoulongne — APPROVED** [review]

**Pull request merged**